src/utils/string.h

```diff
@@ -62,6 +62,7 @@ std::string limitTo(int amount, const std::string &str);
 std::string removeBracketsIfNeeded(std::string a);
 std::string string_to_hex(const std::string& input);
 std::string toHexIfNeeded(const std::string &str);
+std::string toHexIfNeeded(const std::string &str, bool escape_spec);
```

Comment:
Rather than having two (overloaded) functions, with one calling the other, I think this would be a good case for only having a single function but with the second parameter defaulted. E.g. :

toHexIfNeeded(const std::string &str, bool escape_spec = false);

It's not a big deal but it makes the code slightly cleaner and avoids an extra function call on the call stack.

Reply:
Thanks, I re-factorized these methods as you suggested here, see 6dd00be. I keep this review open.